redis-lettuce/src/main/java/io/micronaut/configuration/lettuce/session/RedisSessionStore.java

```diff
@@ -281,9 +281,16 @@ public RedisSession newSession() {
 
     @Override
     public CompletableFuture<Optional<RedisSession>> findSession(String id) {
-        return findSessionInternal(id, false);
+        return findSessionInternal(id, false).thenCompose(session -> {
```

Comment:
Easier to use `thenApply` and just return the `session` inside of wrapping again in `CompletableFuture.completedFuture`

Reply:
updated with proposed changes.